lib/events.js

```diff
@@ -0,0 +1,66 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const Sink = require('fxa-notifier-aws').Sink;
+const P = require('./promise');
+
+const config = require('./config').root();
+const db = require('./db');
+const env = require('./env');
+const logger = require('./logging')('events');
+const workers = require('./img-workers');
+
+const HEX_STRING = require('./validate').hex;
+
+exports.onData = function onData(message) {
+  logger.verbose('data', message);
+  if (message.event === 'delete') {
+    var userId = message.uid.split('@')[0];
+    if (!HEX_STRING.test(userId)) {
+      message.del();
+      return logger.warn('badDelete', { userId: userId });
+    }
+    P.all([
+      db.getAvatars(userId).then(function(avatars) {
+        return P.all(avatars.map(function(avatar) {
+          return workers.delete(avatar.id).then(function() {
+            return db.deleteAvatar(avatar.id);
+          });
+        }));
+      }),
+      db.removeProfile(userId)
+    ])
+      .done(function () {
```

Comment:
maybe nit: something about the indentation here doesn't seem quite right.

We seem to favour `...]).done(...` on a sigle line in this repo, but maybe that doesn't work here because we've got two separate callbacks defined as inline functions?  I don't have a specific alternate suggestion, just wanted to flag it as something that was odd enough to be noticed.